release notes/v0.42.0.md

```diff
@@ -0,0 +1,152 @@
+
+<details>
+<summary> Expand to see an example of the new WebSockets functionality</summary>
+
+This example customizes tags for a WebScoket connection, sets up handlers using the new `on*` syntax, and demonstrates a `ping`/`pong` feature.
+
+```javascript
+import { WebSocket } from "k6/experimental/websockets"
+import { setTimeout, clearTimeout, setInterval, clearInterval } from "k6/experimental/timers"
+
+const CLOSED_STATE = 3
+
+export default function () {
+    var url = "ws://localhost:10000";
+    var params = { "tags": { "my_tag": "hello" } };
+
+    let ws = new WebSocket(url, null, params)
+    
+    ws.onopen = () => {
+        console.log('connected')
+        ws.send(Date.now().toString())        
+    }
+
+    let intervalId = setInterval(() => {
+        ws.ping();
+        console.log("Pinging every 1 sec (setInterval test)")
+    }, 1000);
+
+    let timeout1id = setTimeout(function () {
+        console.log('2 seconds passed, closing the socket')
+        clearInterval(intervalId)
+        ws.close()
+        
+    }, 2000);
+
+    ws.onclose = () => {
+        clearTimeout(timeout1id);
+
+        console.log('disconnected')
+    }
+
+    
+    ws.onping = () => {
+        console.log("PING!")
+    }
+
+    ws.onpong = () => {
+        console.log("PONG!")
+    }
+
+    // Multiple event handlers on the same event
+    ws.addEventListener("pong", () => {
+        console.log("OTHER PONG!")
+    })
+
+    ws.onmessage = (m) => {
+        let parsed = parseInt(m.data, 10)
+        if (Number.isNaN(parsed)) {
+            console.log('Not a number received: ', m.data)
+
+            return
+        }
+
+        console.log(`Roundtrip time: ${Date.now() - parsed} ms`);        
+
+        let timeoutId = setTimeout(function() {
+            if (ws.readyState == CLOSED_STATE) {
+                console.log("Socket closed, not sending anything");
+
+                clearTimeout(timeoutId);
+                return;
+            }
+
+            ws.send(Date.now().toString())
+        }, 500);
+    }   
+
+    ws.onerror = (e) => {
+        if (e.error != "websocket: close sent") {
+            console.log('An unexpected error occurred: ', e.error);
+        }
+    };  
+};
+```
+</details>
+
+The [module docs](https://k6.io/docs/javascript-api/k6-experimental/websockets) have a complete reference, and some examples.
+
+### Wildcard support for `hosts` option [#2747](https://github.com/grafana/k6/pull/2747)
+
+Thanks to the great effort from @eugercek, the [hosts](https://k6.io/docs/using-k6/k6-options/reference/#hosts) option now accepts domains that contain a wildcard at the beginning.
+It can be helpful for setting multiple subdomains of the same domain, so instead of setting `subdomain1.k6.io': '1.2.3.4', 'subdomain2.k6.io': '1.2.3.4'` it is possible to use the wildcard for setting directly `*.test.k6.io: '1.2.3.4'`.
```

Comment:
So that it matches the example.

```suggestion
It can be helpful for setting multiple subdomains of the same domain, so instead of setting `subdomain1.k6.io': '1.2.3.4', 'subdomain2.k6.io': '1.2.3.4'` it is possible to use the wildcard for setting directly `*.k6.io: '1.2.3.4'`.
```